apertium_apy/handlers/translate.py

```diff
@@ -157,26 +157,38 @@ def translate_and_respond(self, pair, pipeline, to_translate, mark_unknown, nosp
         before = self.log_before_translation()
         translated = yield pipeline.translate(to_translate, nosplit, deformat, reformat)
         self.log_after_translation(before, len(to_translate))
-        self.send_response({
+        self.clean_pairs()
+        return {
             'responseData': {
                 'translatedText': self.maybe_strip_marks(mark_unknown, pair, translated),
             },
             'responseDetails': None,
             'responseStatus': 200,
-        })
-        self.clean_pairs()
+        }
 
     @gen.coroutine
     def get(self):
-        pair = self.get_pair_or_error(self.get_argument('langpair'),
-                                      len(self.get_argument('q')))
-        if pair is not None:
-            pipeline = self.get_pipeline(pair)
-            deformat, reformat = self.get_format()
-            yield self.translate_and_respond(pair,
-                                             pipeline,
-                                             self.get_argument('q'),
-                                             self.get_argument('markUnknown', default='yes'),
-                                             nosplit=False,
-                                             deformat=deformat,
-                                             reformat=reformat)
+        query_list = self.get_arguments('q')
+        langpair = self.get_argument('langpair')
+        list_of_futures = []
+        for query in query_list:
+            pair = self.get_pair_or_error(langpair, len(query))
```

Comment:
This line and the next three can go outside the loop, yes?

Reply:
I was thinking of keeping it outside the loop but then I saw that we had to pass the length of the query string to be translated which is used for some logging purposes. Not sure what that is used for though. And other lines after that are dependent on this line, that is why I had to put it inside the loop instead of keeping it outside. What do you suggest?

Reply:
Sending the sum of the lengths should be OK.